### PR TITLE
SDK static builds & update of libs

### PR DESCRIPTION
**Pull request by PeterPetrik**
This PR brings many infrastructure changes, we now use static builds in sdk. A lot of deps have updated version like geodiff 1.x, QGIS 3.22, GDAL 3.4,.. (more in `input-sdk/versions.conf`)

New QGIS version also required to update mapcanvas and add high dpi scaling (https://doc.qt.io/qt-5/highdpi.html).

CI builds will be fixed in following PRs.

### CODE REVIEW

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+            https://github.com/lutraconsulting/input-sdk/releases/download/${{ env.INPUT_SDK_VERSION }}/input-sdk-${{ env.INPUT_SDK_VERSION }}.tar.gz
+          mkdir -p ${{ github.workspace }}/input-sdk
+          cd ${{ github.workspace }}/input-sdk
+          tar -xvzf ${{ github.workspace }}/input-sdk.tar.gz
+
+      # Build Input App
+      - name: Export app/config.pri
+        run: |
+          touch ./input/app/config.pri
+          echo -e "unix:!linuxx:!android {"  >> ./input/app/config.pri
```
Comment (PeterPetrik) [inline .github/workflows/linux.yml:121]: typo ?(linuxx?)

File: app/android.pri [hunk 2]
```diff
@@ -1,49 +1,74 @@
...
+    INCLUDEPATH += $${INPUT_SDK_INCLUDE_PATH}/qgis
+    
+    # by default QMake eats linked libs if they are mentioned multiple times
+    # https://stackoverflow.com/questions/18327959/qmake-how-to-link-a-library-twice/18328971
+    CONFIG += no_lflags_merge
 
-    INCLUDEPATH += $${QGIS_INCLUDE_DIR}
-    LIBS += -L$${QGIS_LIB_DIR}
-    LIBS += -lqgis_core_$${ANDROID_TARGET_ARCH}
+    LIBS += -lc++_shared
```
Comment (PeterPetrik) [inline app/android.pri:28]: is this needed at the end? (remember it was part of our testing)

File: app/inpututils.h [hunk 3]
```diff
@@ -66,6 +66,7 @@ class InputUtils: public QObject
     Q_INVOKABLE double mapSettingsScale( QgsQuickMapSettings *ms );
     Q_INVOKABLE double mapSettingsOffsetX( QgsQuickMapSettings *ms );
     Q_INVOKABLE double mapSettingsOffsetY( QgsQuickMapSettings *ms );
+    Q_INVOKABLE double mapSettingsDPR( QgsQuickMapSettings *ms );
```
Comment (PeterPetrik) [inline app/inpututils.h:69]: why DPR and not DP abbreviation?

File: scripts/docker-build.sh [hunk 4]
```diff
@@ -62,7 +62,7 @@ echo "API: $ANDROIDAPI"
 # Input
 
 # see https://bugreports.qt.io/browse/QTBUG-80756
-export ANDROID_TARGET_ARCH=${ARCH}
+# export ANDROID_TARGET_ARCH=${ARCH}
```
Comment (PeterPetrik) [inline scripts/docker-build.sh:65]: I think this script is now unused, delete

File: scripts/ci/config.pri [hunk 5]
```diff
@@ -36,7 +36,9 @@ ios {
 }
 
 macx:!android {
-  QGIS_INSTALL_PATH = /opt/INPUT/input-sdk-mac-$$(SDK_VERSION)/stage/mac
+  QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.13.0
```
Comment (PeterPetrik) [inline scripts/ci/config.pri:39]: I think this script is now unused, delete

File: qgsquick/qgsquickutils.cpp [hunk 6]
```diff
@@ -34,16 +34,6 @@ qreal QgsQuickUtils::screenDensity() const
...
-  const QWindowList windows = QGuiApplication::topLevelWindows();
-  if ( !windows.isEmpty() )
-  {
-    QScreen *screen = windows.at( 0 )->screen();
-    double dpiX = screen->physicalDotsPerInchX();
-    double dpiY = screen->physicalDotsPerInchY();
-    dpi = dpiX < dpiY ? dpiX : dpiY; // In case of asymmetrical DPI. Improbable
-  }
-  return dpi / 160.;  // 160 DPI is baseline for density-independent pixels in Android
+  // Previously calculated screen pixel density, now kept here for possible future use
```
Comment (PeterPetrik) [inline qgsquick/qgsquickutils.cpp:37]: this is not good...., we need to either remove the function/file altogether from qgsquick (neither we or QField uses it).

we may create a function InputUtils::dp() which will be "scale", or just delete dp alltogether from code.

File: app/main.cpp [hunk 7]
```diff
@@ -327,6 +324,8 @@ void addQmlImportPath( QQmlEngine &engine )
 
 int main( int argc, char *argv[] )
 {
+  QGuiApplication::setAttribute( Qt::AA_EnableHighDpiScaling );
```
Comment (PeterPetrik) [inline app/main.cpp:327]: can you add comment here that QT does autoscale based on dp?

File: app/qml/form/FeatureForm.qml [hunk 8]
```diff
@@ -12,8 +12,6 @@ import QtQuick.Controls 2.0
 import QtQml.Models 2.2
 import QtQml 2.2
 
-// We use calendar in datetime widget that is not yet implemented in Controls 2.2
-import QtQuick.Controls 1.4 as Controls1
```
Comment (PeterPetrik) [inline app/qml/form/FeatureForm.qml:16]: so where is the new calendar? do not see qml here

File: .github/workflows/android.yml [hunk 9]
```diff
@@ -1,58 +1,206 @@
...
+
+      - name: Build armeabi-v7a APK
         env:
-          ARCH: ${{ matrix.arch }}
-          DROPBOX_TOKEN: ${{ secrets.INPUTAPP_BOT_DROPBOX_TOKEN }}
-          GITHUB_TOKEN: ${{ secrets.INPUTAPP_BOT_GITHUB_TOKEN }}
-          INPUTKEYSTORE_STOREPASS: ${{ secrets.INPUTKEYSTORE_STOREPASS }}
+          ARCH: armeabi-v7a
+          VERSION: ${{ env. GIT_TAG }}
+          ANDROID_NDK_ROOT: ${{ steps.setup-ndk.outputs.ndk-path }}
```
Comment (PeterPetrik) [inline .github/workflows/android.yml:183]: so we will use GITHUB_RUN_NUMBER as build number? is this setup?

File: app/inpututils.h (hunk 3 above)
Comment (tomasMizera) [reply app/inpututils.h:69]: DPR = DevicePixelRatio

File: app/qml/form/FeatureForm.qml (hunk 8 above)
Comment (tomasMizera) [reply app/qml/form/FeatureForm.qml:16]: This include was not used anywhere.

Calendar is included from `controls1` in `inputdatetime.qml`, I tested it on multiple devices and it seems that using controls1 does not break the scaling. Calendar, however, have ux issues with hidpi scaling, so I created a ticket to implement a new one once this PR is merged: https://github.com/lutraconsulting/input/issues/1774

File: .github/workflows/android.yml (hunk 9 above)
Comment (tomasMizera) [reply .github/workflows/android.yml:183]: CI changes will come in following PR.

File: app/android.pri (hunk 2 above)
Comment (tomasMizera) [reply app/android.pri]: it is not

**Pull request merged**